src/DIRAC/Resources/Cloud/CloudEndpoint.py

```diff
@@ -0,0 +1,558 @@
+        securityGroupName as stored on the OpenStack service
+
+    :return: S_OK( securityGroup ) | S_ERROR
+    """
+
+    if not securityGroupNames:
+      securityGroupNames = []
+    elif not isinstance(securityGroupNames, list):
+      securityGroupNames = [securityGroupNames]
+
+    if 'default' not in securityGroupNames:
+      securityGroupNames.append('default')
+
+    try:
+      secGroups = self.__driver.ex_list_security_groups()
+      # the libcloud library, throws Exception. Nothing to do.
+    except Exception as errmsg:
+      return S_ERROR(errmsg)
+
+    return S_OK([secGroup for secGroup in secGroups if secGroup.name in securityGroupNames])
+
+  def createInstances(self, vmsToSubmit):
+    outputDict = {}
+
+    for nvm in range(vmsToSubmit):
+      instanceID = makeGuid()[:8]
+      createPublicIP = 'ipPool' in self.parameters
+      result = self.createInstance(instanceID, createPublicIP)
+      if result['OK']:
+        node, publicIP = result['Value']
+        self.log.debug('Created VM instance %s/%s with publicIP %s' % (node.id, instanceID, publicIP))
+        nodeDict = {}
+        nodeDict['PublicIP'] = publicIP
+        nodeDict['InstanceID'] = instanceID
+        nodeDict['NumberOfProcessors'] = self.flavor.vcpus
+        nodeDict['RAM'] = self.flavor.ram
+        nodeDict['DiskSize'] = self.flavor.disk
+        nodeDict['Price'] = self.flavor.price
+        outputDict[node.id] = nodeDict
+      else:
+        break
+
+    # We failed submission utterly
+    if not outputDict:
+      return result
+
+    return S_OK(outputDict)
+
+  def createInstance(self, instanceID='', createPublicIP=True):
+    """
+    This creates a VM instance for the given boot image
+    and creates a context script, taken the given parameters.
+    Successful creation returns instance VM
+
+    Boots a new node on the OpenStack server defined by self.endpointConfig. The
+    'personality' of the node is done by self.imageConfig. Both variables are
+    defined on initialization phase.
+
+    The node name has the following format:
+    <bootImageName><contextMethod><time>
+
+    It boots the node. If IPpool is defined on the imageConfiguration, a floating
+    IP is created and assigned to the node.
+
+    :return: S_OK( ( nodeID, publicIP ) ) | S_ERROR
+    """
+
+    if not instanceID:
+      instanceID = makeGuid()[:8]
+
+    self.parameters['VMUUID'] = instanceID
+    self.parameters['VMType'] = self.parameters.get('CEType', 'OpenStack')
+
+    createNodeDict = {}
+
+    # Get the image object
+    if "ImageID" in self.parameters and 'ImageName' not in self.parameters:
+      result = self.__getImageByName(self.parameters['ImageName'])
+      if not result['OK']:
+        return result
+      image = result['Value']
+    elif "ImageID" in self.parameters:
+      try:
+        image = self.__driver.get_image(self.parameters['ImageID'])
+      except Exception as e:
+        if "Image not found" in str(e):
```

Comment:
I am sure there's a more precise way.

Reply:
Yes, converted to use the proper exception type and check the numeric error code for the special not found case.